feat(kurtosis-devnet): extract information from op-deployer artifact

sigma opened the pull request:
**Description**

This is in service of outputting useful information at the end of the
deployment.

Here we get back some useful wallets and addresses for downstream
consumption.

**Additional context**

gradually porting changes from
https://github.com/ethereum-optimism/optimism/tree/sigma/kurtosis-devnet-rebase/kurtosis-devnet

**Metadata**

Part of ethereum-optimism/platforms-team#519

sigma committed:
feat(kurtosis-devnet): extract information from op-deployer artifact

This is in service of outputting useful information at the end of the
deployment.

Here we get back some useful wallets and addresses for downstream
consumption.

protolambda commented on kurtosis-devnet/pkg/kurtosis/sources/deployer/wallets.go:
```diff
@@ -0,0 +1,106 @@
+package deployer
+
+// TODO: we should be getting this from somewhere in the enclave. It's ok-ish
+// though, as this is hard-coded in ethereum-optimism anyway.
+var knownWallets = []*Wallet{
+	{
```
There's a `op-chain-ops/devkeys` package that may be useful for dev purposes, where keys are set up all from the same mnemonic, in a structured way, for all chains / roles.

sigma replied:
ah cool. Will take a look

sigma replied:
Actually, do you mind if I do that in a followup PR?
I figured I should probably use the same for  the operators keys, but I need to check I'm getting/using the right mnemonics/roles/... which will be easier once I land a workable end-to-end tool

sigma replied:
Nevermind, I started using devkeys for the users.
We'll  see later what other keys are useful

sigma commented:
I rewrote the code to make use of the kurtosis SDK instead of shelling out.
Also introduced a "filesystem" abstraction on top of artifacts, for convenience
